src/DynamoCore/Core/DelayedGraphExecution.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Threading;
+using Dynamo.Graph.Workspaces;
+
+namespace Dynamo.Core
+{
+    /// <summary>
+    /// This class enables the delay of graph execution.
+    /// Use instances of this class to specify a code scope in which you want graph execution to be delayed. 
+    /// Class is thread safe. 
```

Comment:
hmm - it might be threadSafe - but does it have well defined behavior?

Reply:
no, probably not. We can specify that.
And improve functionality in the future as the need requires (both for multi thread and nested-ness behavior)

Reply:
thats fine with me, can we note it in the summary.